src/core/src/service_interfaces/StatusHandler.py

```diff
@@ -795,14 +835,402 @@ def __try_add_error(error_list, detail):
+        byte_size[list[:1]] < byte_size[list[:2]] < byte_size[list[:3]] ...
+        return truncated_list, packages_removed_from_list, and remaining max_package_list_capacity
+        """
+        left_index = 0
+        right_index = len(package_list) - 1
+
+        if len(package_list) == 0:
+            return [], [], capacity
+        if self.__calc_package_payload_size_on_disk(package_list) <= capacity:
+            return package_list, [], capacity - self.__calc_package_payload_size_on_disk(package_list)
+        if self.__calc_package_payload_size_on_disk(package_list[0]) > capacity:
+            return [], package_list, capacity
+
+        while left_index < right_index:
+            mid_index = left_index + int((right_index - left_index) / 2)
+            if self.__calc_package_payload_size_on_disk(package_list[:mid_index]) >= capacity:
+                right_index = mid_index
+            else:
+                left_index = mid_index + 1
+
+        truncated_list = package_list[:left_index - 1]
+        packages_removed_from_list = package_list[left_index - 1:]
+        return truncated_list, packages_removed_from_list, capacity - self.__calc_package_payload_size_on_disk(truncated_list)
+
+    def __removed_older_complete_status_files(self, status_folder):
+        """ Retain 10 latest status complete file and remove other .complete.status files """
+        files_to_removed = []
+        complete_status_files_list = glob.glob(status_folder + '/' + '*.complete.status')   # Glob return empty list if no file matched pattern
+        complete_status_files_list.sort(key=os.path.getmtime, reverse=True)
+
+        if len(complete_status_files_list) <= 10:
+            return
+
+        for file in complete_status_files_list[10:]:
+            try:
+                if os.path.exists(file):
+                    os.remove(file)
+                    files_to_removed.append(file)
+            except Exception as e:
+                self.composite_logger.log_debug("Error deleting complete status file. [File={0} [Exception={1}]]".format(repr(file), repr(e)))
+
+        self.composite_logger.log_debug("Cleaned up older complete status files: {0}".format(files_to_removed))
+
+    def __calc_status_size_on_disk(self, full_status):
+        """ Calculate status file size in bytes on disk """
+        return len(json.dumps(full_status).encode("utf-8"))
+
+    def __calc_package_payload_size_on_disk(self, package_list):
+        """ Calculate final package list size in bytes (because of escape chars) """
+        first_json_dump = json.dumps(package_list)
+        return len(json.dumps(first_json_dump).encode("utf-8"))
+
+    def __get_size_of_constant_status_data(self, complete_status_file_payload, assessment_status_index, installation_status_index):
+        """ Get the size in bytes of the complete_status_file without package data  """
+        status_file_no_list_data = complete_status_file_payload
```

Comment:
If complete_status_file_payload is a deep copy from the caller, what is the need to assign it to a new variable here?

Reply:
Use the name reformatted_status_file instead of complete_status_file_payload or status_file_no_list_data

Reply:
it was done per koshy's suggestion to help code readability, but